### ISSUE TITLE
Property set to mandatory and using MNTP validates as invalid inside block

### ISSUE THREAD

**Issue by bjarnef** [umbraco/Umbraco-CMS#14484]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

11.4.1

### Bug summary

I added a property using MTNP inside a block inserted in Block Grid editor and the property is configured as mandatory, but on save it validates as invalid even though a node is selected.

![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/93e9e016-f836-4ebe-8b4d-2090fd12129b)

### Steps to reproduce

Add a property to a block element using MNTP and set property to mandatory.
Insert the block in Block Grid and try to save or save & publish node.

**Comment by elit0451** [comment]
Hey @bjarnef 👋 

Thanks for reaching out. Unfortunately, I can't reproduce the issue. Can you have a look at this video and let me know what I am missing?

https://github.com/umbraco/Umbraco-CMS/assets/21998037/791a40ab-442d-4561-b57c-d6305b62167c

**Comment by bjarnef** [comment]
Hi @elit0451 

Odd I could reproduce it again - this time inside Block List.

https://github.com/umbraco/Umbraco-CMS/assets/2919859/08a0a2f9-7856-4ca6-9e1b-9b8c7ede8880

It seems to happen when I have existing content and and added a property "Site" using MNTP and set as mandatory.

Hitting save & publish then mark errors at all items, since none of them have value in "Site" property, but selecting a value doesn't invalidate the property on first attempt - I have to remove the selected item and select it once again and then the error disappears from the property and I can submit/close the overlay.

**Comment by elit0451** [comment]
I managed to reproduce this now - it happens when you have existing content for the element type and you add **ANY** mandatory field. Saving and publishing will result in the `"Value cannot be null"` validation error even when you have filled out the mandatory field.

<img width="1133" alt="Screenshot 2023-07-05 at 08 30 47" src="https://github.com/umbraco/Umbraco-CMS/assets/21998037/09c52e53-2201-4366-a796-7cd0a0d4bde2">


</br>
</br>
I will mark this as up for grabs in case someone wants to give it a try. The workaround is to add the property as a non-mandatory, fill out the content and then make it mandatory.

**Comment by bjarnef** [comment]
Yes, it with MNTP the workaround is to remove the item selected first time and select it again. On second selection the validation is updated the be valid.
